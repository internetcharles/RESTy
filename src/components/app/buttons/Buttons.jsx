import React from 'react';
import PropTypes from 'prop-types';

const Buttons = ({ onChange }) => (
  <div 
    onChange={onChange}
  >
    <input type="radio" value="GET" name="get">GET</input>
    <input type="radio" value="POST" name="post">GET</input>
    <input type="radio" value="PUT" name="put">GET</input>
    <input type="radio" value="DELETE" name="delete">GET</input>
    <textarea name="data" onChange={onChange} />
  </div>
);

Buttons.propTypes = {
  onChange: PropTypes.func.isRequired
};

export default Buttons;
