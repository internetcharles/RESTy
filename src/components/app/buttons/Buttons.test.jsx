import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Buttons from './Buttons';

describe('Buttons component', () => {
  afterEach(() => cleanup());
  it('renders buttons', () => {
    const { asFragment } = render(<Buttons
      onChange={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
