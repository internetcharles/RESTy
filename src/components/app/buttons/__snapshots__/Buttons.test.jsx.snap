// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons component renders buttons 1`] = `
<DocumentFragment>
  <div>
    <input
      name="get"
      type="radio"
      value="GET"
    />
    <input
      name="post"
      type="radio"
      value="POST"
    />
    <input
      name="put"
      type="radio"
      value="PUT"
    />
    <input
      name="delete"
      type="radio"
      value="DELETE"
    />
    <textarea
      name="data"
    />
  </div>
</DocumentFragment>
`;
